tests/TestCase.php

```diff
@@ -206,57 +208,84 @@ protected function getCollectionName(): string
     /**
      * Return a list of invalid array values.
      */
-    protected function getInvalidArrayValues(bool $includeNull = false): array
+    protected static function getInvalidArrayValues(bool $includeNull = false): array
     {
         return [123, 3.14, 'foo', true, new stdClass(), ...($includeNull ? [null] : [])];
     }
 
     /**
      * Return a list of invalid boolean values.
      */
-    protected function getInvalidBooleanValues(bool $includeNull = false): array
+    protected static function getInvalidBooleanValues(bool $includeNull = false): array
     {
         return [123, 3.14, 'foo', [], new stdClass(), ...($includeNull ? [null] : [])];
     }
 
     /**
      * Return a list of invalid document values.
      */
-    protected function getInvalidDocumentValues(bool $includeNull = false): array
+    protected static function getInvalidDocumentValues(bool $includeNull = false): array
     {
         return [123, 3.14, 'foo', true, PackedArray::fromPHP([]), ...($includeNull ? [null] : [])];
     }
 
-    protected function getInvalidObjectValues(bool $includeNull = false): array
+    protected static function getInvalidObjectValues(bool $includeNull = false): array
     {
         return [123, 3.14, 'foo', true, [], new stdClass(), ...($includeNull ? [null] : [])];
     }
 
-    protected function getInvalidDocumentCodecValues(): array
+    protected static function getInvalidDocumentCodecValues(): array
     {
-        return [123, 3.14, 'foo', true, [], new stdClass(), $this->createMock(Codec::class)];
+        $codec = new class implements Codec {
+            use DecodeIfSupported;
+            use EncodeIfSupported;
+
+            public function canDecode(mixed $value): bool
+            {
+                return true;
+            }
+
+            public function decode(mixed $value): mixed
+            {
+                return $value;
+            }
+
+            public function canEncode(mixed $value): bool
+            {
+                return true;
+            }
+
+            public function encode(mixed $value): mixed
+            {
+                return $value;
+            }
+        };
+        // @fixme: createStub can be called statically in PHPUnit 10
+        // $codec = self::createStub(Codec::class);
```

Comment:
I assume once you upgrade the PHPUnit 10, the above anonymous class can be replaced with the single `createStub()` call?

Should this be tracked in a sub-task within [PHPLIB-1369](https://jira.mongodb.org/browse/PHPLIB-1369), or do you plan on addressing it in the main ticket?

Reply:
Tracked: https://jira.mongodb.org/browse/PHPLIB-1516